packages/carousel/src/index.js

```diff
@@ -0,0 +1,228 @@
+    })
+  }
+
+  updateSelectedItem(selectedItemIndex) {
+    this.setState({
+      internalCurrentItemIndex: selectedItemIndex
+    })
+  }
+
+  mapItems() {
+    const { items } = this.state
+
+    const currentItemIndex =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+
+    const itemsToShow = []
+    items.forEach((item, index) => {
+      if (index <= currentItemIndex + 2) {
+        itemsToShow.push(
+          <span key={`item-${index}`}>
+            <PositionItem>{item}</PositionItem>
+          </span>
+        )
+      }
+    })
+
+    return itemsToShow
+  }
+
+  mapDots() {
+    const currentItemIndex =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+
+    return this.state.items.map((item, mapIndex) => (
+      <Dot
+        key={mapIndex}
+        mx={2}
+        mt={2}
+        bg={mapIndex === currentItemIndex ? 'blue' : 'borderGray'}
+        onClick={this.updateSelectedItem.bind(this, mapIndex)}
+      />
+    ))
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      items: nextProps.children
+    })
+  }
+
+  render() {
+    const { currentItemTranslate, isSwiping } = this.state
+
+    const { shouldShowArrows } = this.props
+
+    const currentItemIndex =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+
+    return (
+      <Box {...this.props}>
+        <OverflowContainer w={1} mb={3}>
+          {shouldShowArrows && (
+            <div>
+              <PositionedArrowLeftController
+                handleClick={this.handleSwipedRight}
+                iconDirection="chevronLeft"
+              />
+              <PositionedArrowRightController
+                handleClick={this.handleSwipedLeft}
+                iconDirection="chevronRight"
+              />
+            </div>
+          )}
+          <Swipeable
+            onSwipingLeft={this.handleSwipingLeft}
+            onSwipingRight={this.handleSwipingRight}
+            onSwipedLeft={this.handleSwipedLeft}
+            onSwipedRight={this.handleSwipedRight}
+            trackMouse
+          >
+            <SwiperWrap
+              translateX={-92 * currentItemIndex + currentItemTranslate + 4}
+              isSwiping={isSwiping}
+            >
+              {this.mapItems()}
+            </SwiperWrap>
+          </Swipeable>
+        </OverflowContainer>
+        <Flex wrap justify="center">
+          {this.mapDots()}
+        </Flex>
+      </Box>
+    )
+  }
+}
+
+const OverflowContainer = styled(Relative)`
+  overflow: hidden;
+`
+
+const Dot = styled(Box)`
+  height: 12px;
+  width: 12px;
+  border-radius: 50%;
+  cursor: pointer;
+`
+
+Dot.displayName = 'Dot'
+
+const PositionedArrowLeftController = styled(ArrowController)`
+  position: absolute;
+  top: 38%;
+  z-index: 1;
+`
+
+const PositionedArrowRightController = styled(PositionedArrowLeftController)`
+  right: 0;
+`
+
+const SwiperWrap = styled.div`
+  transition-property: all;
```

Comment:
This can be expensive to transition all properties, rather than something like `transform`